### PR TITLE
New loss function for Neural Networks based on 2018 paper from Cornell 

### PR DESCRIPTION
**Pull request by paulobuchsbaum**
Zhilu Zhang and Mert R. Sabuncu of Cornell University (New York, United States) has submitted on May 20, 2018 a [paper](https://arxiv.org/abs/1805.07836)  proposing a new error function for `Neural Networks` (including Deep Neural Networks) aimed at classification with `softmax` function in the output layer,  specially in the case of noisy labels. 

This new loss function has joined characteristics of `mean absolute error (MAE)` with `cross entropy loss`. 

According to the tests carried out, it generates a significant improvement in problems of binary classification or multiclassification with noisy labels, compared to cross entropy loss or mean absolute error (MAE) 

By e-mail, we exchanged ideas for some time, where I has clarified some points of this paper. 

Because of the promising results obtained, I've agreed with them that I would incorporate this new function into the repertoire of loss functions in `TensorFlow` to make it available to the community.

So I've read the `contribution guidelines` and write Python code for new  function (that I've called  `gen_crossentropy`) and also write the testing code, following the standards of another loss error functions. 

In my local repository, the test have run 100% well. 

It's my first collaboration with this fantastic project, so any observations or guidance will be welcome.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: It has been 14 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by paulobuchsbaum** [comment]
I'm back and waiting ...

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by fchollet** [comment]
Thank you for the PR.

After review, we have decided not to make this addition to the Keras API.

Every feature has a cost that goes beyond the initial PR: maintenance cost, documentation cost, and cognitive cost for users. In the Keras API, every new feature has to be maintained in perpetuity, and has to be replicated in every implementation of the Keras API (which includes tf.keras, multi-backend Keras, tensorflow.js, keras-mxnet, and others).

As, such, our criteria for adding a new feature in the API is the following:

- It should be broadly useful to Keras users, not a niche feature. Niche features should be maintained independently by those who need them (e.g. by extending the API via subclassing), as third-party Keras add-on packages.
- It should be broadly recognized as a deep learning best practice. We don’t add new layers/etc that were recently published to ArXiv.org, we only add new objects that are already commonly used in the deep learning community.
- It should have an owner committed to maintaining it in the long term.
